crates/core/src/workflows/get_link_result.rs

```diff
@@ -10,11 +11,20 @@ use crate::{
 };
 
 use holochain_core_types::error::HolochainError;
+use holochain_persistence_api::cas::content::{Address, AddressableContent};
 use holochain_wasm_utils::api_serialization::get_links::{
     GetLinksArgs, GetLinksResult, LinksResult,
 };
 use std::sync::Arc;
 
+// check to see if we are an authority on the DHT for this base, if so no need
+// to go out to request this from anybody else.  We know for sure that we are an authority
+// for anything that is linked on our own agent hash.
+pub fn am_i_dht_authority_for_base<'a>(context: &'a Arc<Context>, base: &Address) -> bool {
+    let me: Address = context.agent_id.address();
+    *base == me
```

Comment:
Hm, this actually does not check if we are a DHT authority for the base, which we easily could by testing if base is included in the holding list. Shouldn‘t we do that here?

Reply:
added